File: src/main/java/com/sap/cx/swell/services/WorkflowDefService.java
package com.sap.cx.swell.services;

import com.sap.cx.swell.model.worflowdef.WorkflowDef;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface WorkflowDefService {

    Mono<WorkflowDef> create(WorkflowDef e);

    Mono<WorkflowDef> findById(String id);

    Mono<WorkflowDef> update(WorkflowDef e);

    Mono<Void> delete(String id);
}

File: lib/swell/services/workflow_def_service.ex
defmodule Swell.Services.WorkflowDefService do
  alias Swell.Repos.WorkflowDefRepo

  def get_with_id(id) do
    WorkflowDefRepo.find_by_id(id)
  end

  def create(workflow_def) do
    WorkflowDefRepo.create(workflow_def)
  end

  def update(id, workflow_def) do
    WorkflowDefRepo.update(id, workflow_def)
  end

  def delete(id) do
    WorkflowDefRepo.delete(id)
  end
  
end
